### PR TITLE
Resolved Code Analysis Vulnerabilities 

### PR DESCRIPTION
**Pull request by sedin-tushar**
# SunbirdEd - Portal
---

### Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
---

#### Example
- [x] Applicable
- [ ] Not applicable

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/client/package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
     "@angular/compiler-cli": "14.3.0",
     "@angular/core": "14.3.0",
     "@angular/forms": "14.3.0",
-    "@angular/localize": "14.3.0",
+    "@angular/localize": "16.0.6",
```
Comment (princegupta1131) [inline src/app/client/package.json:51]: @sedin-tushar please don't change any "@angular" related upgrade

File: src/app/client/package.json [hunk 2]
```diff
@@ -39,7 +39,7 @@
   },
   "private": true,
   "dependencies": {
-    "@angular-devkit/build-angular": "14.2.10",
+    "@angular-devkit/build-angular": "14.2.12",
```
Comment (princegupta1131) [inline src/app/client/package.json:42]: @sedin-tushar please don't change any "https://github.com/angular" related upgrade

File: src/app/client/package.json [hunk 3]
```diff
@@ -77,7 +77,7 @@
     "@project-sunbird/sunbird-quml-player-v9": "5.1.5",
     "@project-sunbird/sunbird-resource-library": "7.0.3",
     "@project-sunbird/sunbird-video-player-v9": "5.5.1",
-    "@project-sunbird/telemetry-sdk": "0.0.29",
+    "@project-sunbird/telemetry-sdk": "1.3.0",
```
Comment (princegupta1131) [inline src/app/client/package.json:80]: @sedin-tushar did u test the telemetry related functionality post this changes

File: src/app/routes/clientRoutes.js [hunk 4]
```diff
@@ -284,11 +284,13 @@ const renderTenantPage = (req, res) => {
 }
 // in fallback option check always for local tenant folder and redirect to / if not exists
 const loadTenantFromLocal = (req, res) => {
-  const tenantName = _.lowerCase(req.params.tenantName) || envHelper.DEFAULT_CHANNEL
-  if (tenantName && fs.existsSync(path.join(__dirname, './../tenant', tenantName, 'index.html'))) {
-    res.sendFile(path.join(__dirname, './../tenant', tenantName, 'index.html'))
+  let tenantName = _.lowerCase(req.params.tenantName) || envHelper.DEFAULT_CHANNEL;
```
Comment (princegupta1131) [inline src/app/routes/clientRoutes.js:287]: @sedin-tushar why u have added this code? please add the description

File: src/desktop/modules/controllers/batchDetails.ts [hunk 5]
```diff
@@ -42,7 +43,7 @@ export default class BatchDetails {
 
       if (_.get(response, 'docs.length')) {
         batchData = response.docs[0];
-        res.status(200).send(Response.success(API_ID, { response: batchData }, req));
+        res.status(200).send(xss(JSON.stringify(Response.success(API_ID, { response: batchData }, req))));
```
Comment (princegupta1131) [inline src/desktop/modules/controllers/batchDetails.ts:46]: @sedin-tushar why u have added this code? please add the description

**Review by princegupta1131 — COMMENTED** [review]
const xss = require('xss'); why u have added this import

File: src/desktop/modules/controllers/batchDetails.ts (hunk 5 above)
Comment (sedin-tushar) [reply src/desktop/modules/controllers/batchDetails.ts:46]: should i add Description in each file ? there are about 10 files with this change

**Comment by sedin-tushar** [comment]
> const xss = require('xss'); why u have added this import

The xss package is used for preventing cross-site scripting (XSS) attacks by sanitizing user input , thus reducing High Impact Xss Vulnerabilities

File: src/app/client/package.json (hunk 3 above)
Comment (mithun30052001) [reply src/app/client/package.json:80]: How can  i specifically check telemetry-functionalities @princegupta1131

Comment (mithun30052001) [reply src/app/client/package.json]: The telemetry-sdk library is downgraded to previous version itself

File: src/app/client/package.json (hunk 1 above)
Comment (mithun30052001) [reply src/app/client/package.json]: The @angular versions are downgraded to previous versions

**Review by rajeevsathish — APPROVED** [review]

**Pull request merged**